WordPress/gutenberg
# Full width alignments and root padding combined.

## What?
This combines https://github.com/WordPress/gutenberg/pull/39871 and https://github.com/WordPress/gutenberg/pull/39926  which makes it clear that there are some shortcomings with the approach in these PRs.

The problem is that the root padding PR assumes that blocks that implement "inherit default layout" will always be at the root of the document, so it uses `.wp-site-blocks > * > .alignfull` to target them.

However this is not always the case - the current behaviour in trunk is that alignfull items inside the outermost block in the document that implements `inherit default layout` will stretch to the edge of the screen.

I have made some adjustments so that we can achieve the aims of both PRs...

## Why?
So we can remove custom CSS from block themes.

## How?
This is difficult to achieve, because we want the padding set on the root element, but the negative margins set on the outermost `.alignfull` block. The approach here is to set the negative margins to counteract the root padding on _all_ `.alignfull` blocks, and then to override this with more specific negative margins for inner `alignfull` blocks.

To achieve this it was necessary to add a utility class (`.wp-containter-default-layout`) to all blocks that inherit the default layout, so that we can target those that are more than one level deep. I appreciate this is convoluted, but it's the only way I could get this to work.

Another benefit of the approach here is that we don't have to force blocks that implement "inherit default layout" to use the root padding - they can have their own padding values, which should work, and the alignfull blocks should adjust for this. I haven't yet implemented this, but I believe it will work.

Right now this is only working in the frontend

Still to do:
-[ ] Remove the code that prevents users from changing padding on "inherit default layout" blocks
-[ ] Update WP_Theme_JSON to use the set padding values if they are defined
-[ ] Move WP_Theme_JSON changes to the 6.1 compat dir
-[ ] Update the CSS in the editor to match the changes in the frontend.

## Testing Instructions
- Use this PR: https://github.com/Automattic/themes/pull/5949
- Switch to Archeo
- You can also use this block markup to test: https://gist.github.com/scruffian/df0ce92c85cffdfee0d69fbe2bcb8995
- Check that the alignments work as expected

## Screenshots or screencast 
TODO

@WordPress/block-themers
